File: jose.4/candidatos/otroo/celular.php
<?php
session_start();

// Inicializar las variables de sesión si aún no existen
if (!isset($_SESSION['ventas'])) {
    $_SESSION['ventas'] = array(
        'samsung' => 0,
        'motorola' => 0,
        'huawei' => 0
    );
}

// Definir los precios de los celulares
$precios = array(
    'samsung' => 600000,
    'motorola' => 500000,
    'huawei' => 550000
);

// Función para aplicar el descuento
function calcularDescuento($cantidad) {
    return ($cantidad > 10) ? 0.05 : 0;
}

// Procesar el formulario de ventas
if (isset($_POST['agregarVenta'])) {
    $marca = $_POST['marca'];
    $cantidad = (int)$_POST['cantidad'];

    if ($cantidad <= 0) {
        $error = 'Ingrese una cantidad válida.';
    } else {
        $_SESSION['ventas'][$marca] += $cantidad;
    }
}
?>

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Ventas de Celulares</title>
</head>
<body>
    <h1>Ventas de Celulares</h1>
    <form method="post">
        <label for="marca">Seleccione una marca:</label>
        <select id="marca" name="marca">
            <option value="samsung">Samsung</option>
            <option value="motorola">Motorola</option>
            <option value="huawei">Huawei</option>
        </select>
        <br>
        <label for="cantidad">Cantidad vendida:</label>
        <input type="number" id="cantidad" name="cantidad" min="1" required>
        <br>
        <button type="submit" name="agregarVenta">Agregar Venta</button>
    </form>

    <?php
    if (isset($error)) {
        echo '<p>' . $error . '</p>';
    }

    // Mostrar resultados
    echo '<h2>Resultado:</h2>';
    foreach ($_SESSION['ventas'] as $marca => $cantidadVendida) {
        $precioUnitario = $precios[$marca];
        $subtotal = $cantidadVendida * $precioUnitario;
        $descuentoAplicado = calcularDescuento($cantidadVendida) * $subtotal;
        $total = $subtotal - $descuentoAplicado;

        echo '<h3>' . ucfirst($marca) . '</h3>';
        echo '<p>Cantidad Vendida: ' . $cantidadVendida . '</p>';
        echo '<p>Precio Unitario: ' . $precioUnitario . '</p>';
        echo '<p>Subtotal: ' . $subtotal . '</p>';
        echo '<p>Descuento Aplicado: ' . $descuentoAplicado . '</p>';
        echo '<p>Total: ' . $total . '</p>';
    }
    ?>
</body>
</html>

File: jose.4/candidatos/otroo/celular.py
# Inicializar variables de ventas por marca
ventas = {
    'samsung': 0,
    'motorola': 0,
    'huawei': 0
}

# Definir precios de celulares
precios = {
    'samsung': 600000,
    'motorola': 500000,
    'huawei': 550000
}

def calcular_descuento(cantidad):
    return 0.05 if cantidad > 10 else 0

def mostrar_resultados():
    print("\nResultados de Ventas:")
    for marca, cantidad in ventas.items():
        precio_unitario = precios[marca]
        subtotal = cantidad * precio_unitario
        descuento = calcular_descuento(cantidad)
        descuento_aplicado = subtotal * descuento
        total = subtotal - descuento_aplicado

        print(f"Marca: {marca.capitalize()}")
        print(f"Cantidad Vendida: {cantidad}")
        print(f"Precio Unitario: {precio_unitario}")
        print(f"Subtotal: {subtotal}")
        print(f"Descuento: {descuento * 100}%")
        print(f"Descuento Aplicado: {descuento_aplicado}")
        print(f"Total: {total}\n")

# Ciclo para ingresar ventas
for _ in range(1):
    print("Seleccione una marca:")
    print("1. Samsung")
    print("2. Motorola")
    print("3. Huawei")
    
    opcion = int(input())
    
    if opcion < 1 or opcion > 3:
        print("Opción no válida.")
        continue
    
    marca = None
    if opcion == 1:
        marca = 'samsung'
    elif opcion == 2:
        marca = 'motorola'
    elif opcion == 3:
        marca = 'huawei'
    
    cantidad = int(input("Ingrese la cantidad vendida: "))
    
    if cantidad <= 0:
        print("Ingrese una cantidad válida.")
    else:
        ventas[marca] += cantidad

mostrar_resultados()
